go-controller/pkg/testing/testifymockhelper.go

```diff
@@ -1,7 +1,60 @@
 package testing
 
+import (
+	"github.com/stretchr/testify/mock"
+)
+
+// TestifyMockHelper captures the arguments needed for the `On` , `Return` and 'Times` method ( refer
+// https://godoc.org/github.com/stretchr/testify/mock#Call.On ,
+// https://godoc.org/github.com/stretchr/testify/mock#Call.Return and
+// https://godoc.org/github.com/stretchr/testify/mock#Call.Times )
 type TestifyMockHelper struct {
-	OnCallMethodName    string
+	// OnCallMethodName - mock method that will be called.
+	// Refer the `Method` field at https://godoc.org/github.com/stretchr/testify/mock#Call
+	OnCallMethodName string
+	// OnCallMethodArgType - argument types of the method that will be called.
+	// Refer the `Arguments` field at https://godoc.org/github.com/stretchr/testify/mock#Call
 	OnCallMethodArgType []string
-	RetArgList          []interface{}
+	// RetArgList - arguments returned by mock method when called.
+	// Refer the `ReturnArguments` field at https://godoc.org/github.com/stretchr/testify/mock#Call
+	RetArgList []interface{}
+	// OnCallMethodsArgsStrTypeAppendCount - number of times the `string` type argument is repeated at the end.
+	// NOTE: There are a couple of cases where the ovn related calls take upto 23 string arguments
+	OnCallMethodsArgsStrTypeAppendCount int
+	// CallTimes - number of times to return the return arguments when mock method is called
+	// Refer the `Repeatability` field at https://godoc.org/github.com/stretchr/testify/mock#Call
+	// NOTE: The zero value for an int is 0 in go, however, it the value for 'CallTimes' is not set, we interpret that
+	// it will need to be called once.
+	CallTimes int
```

Comment:
The kind of change I am proposing might need another PR because of the size but what about changing the names of these struct fields and maybe the struct name? Something that makes it more obvious to the reader that has no idea what is going on what these fields are something like 
```
type ExpectedFunctions struct {
  Name string
  ArgTypes [] string
  FuncReturns: []interface{}
  CallTimes int
}
```
My reasoning is that Testify and Mocking don't really mean anything to the person writing or reading the tests and these names I feel would make the tests more approachable

Reply:
Great suggestion, I shall open an issue to track to be revisited in a later PR .
Refer https://github.com/ovn-org/ovn-kubernetes/issues/1854